internal/services/v1/permissions.go

```diff
@@ -94,21 +98,18 @@ func (ps *permissionServer) CheckPermission(ctx context.Context, req *v1.CheckPe
 	)
 	usagemetrics.SetInContext(ctx, metadata)
 
+	var debugTrace *v1.DebugInformation
 	if debugOption != computed.NoDebugging && metadata.DebugInfo != nil {
-		// Convert the dispatch debug information into API debug information and marshal into
-		// the footer.
+		// Convert the dispatch debug information into API debug information.
 		converted, cerr := ConvertCheckDispatchDebugInformation(ctx, caveatContext, metadata, ds)
 		if cerr != nil {
 			return nil, ps.rewriteError(ctx, cerr)
 		}
+		debugTrace = converted
 
-		marshaled, merr := protojson.Marshal(converted)
-		if merr != nil {
-			return nil, ps.rewriteError(ctx, merr)
-		}
-
+		// TODO(jschorr): Remove this once the trailer is not longer used by anyone.
 		serr := responsemeta.SetResponseTrailerMetadata(ctx, map[responsemeta.ResponseMetadataTrailerKey]string{
-			responsemeta.DebugInformation: string(marshaled),
+			responsemeta.DebugInformation: `{"note": "debug information has been moved into the response object."}`,
```

Comment:
Maybe this should link to the docs page on clients or something and a note that you need to upgrade the client to see those messages?

Reply:
I could add a spicedb.dev link?

Reply:
SGTM

Reply:
Updated